src/ResourceManagement/CosmosDB/Domain/SqlContainer/Definition/IDefinition.cs

```diff
@@ -0,0 +1,186 @@
+        IWithAttach<ParentT> WithDefaultTtl(int ttl);
+    }
+
+    /// <summary>
+    /// The stage of a sql container definition allowing to set unique key policy.
+    /// </summary>
+    /// <typeparam name="ParentT">The stage of the parent definition to return to after attaching this definition.</typeparam>
+    public interface IWithUniqueKeyPolicy<ParentT>
+    {
+        /// <summary>
+        /// Specifies the unique key policy.
+        /// </summary>
+        /// <param name="uniqueKeyPolicy">The whole object of the unique key policy.</param>
+        /// <returns>The next stage of the definition.</returns>
+        IWithAttach<ParentT> WithUniqueKeyPolicy(UniqueKeyPolicy uniqueKeyPolicy);
+
+        /// <summary>
+        /// Specifies the list of unique key.
+        /// </summary>
+        /// <param name="uniqueKeys">The list of unique key.</param>
+        /// <returns>The next stage of the definition.</returns>
+        IWithAttach<ParentT> WithUniqueKeys(IList<UniqueKey> uniqueKeys);
+
+        /// <summary>
+        /// Specifies a unique key appended to original list.
+        /// </summary>
+        /// <param name="uniqueKey">A unique key.</param>
+        /// <returns>The next stage of the definition.</returns>
+        IWithAttach<ParentT> WithUniqueKey(UniqueKey uniqueKey);
+    }
+
+    /// <summary>
+    /// The stage of a sql container definition allowing to set conflict resolution policy.
+    /// </summary>
+    /// <typeparam name="ParentT">The stage of the parent definition to return to after attaching this definition.</typeparam>
+    public interface IWithConflictResolutionPolicy<ParentT>
+    {
+        /// <summary>
+        /// Specifies the conflict resolution policy.
+        /// </summary>
+        /// <param name="conflictResolutionPolicy">The whole object of the conflict resolution policy.</param>
+        /// <returns>The next stage of the definition.</returns>
+        IWithAttach<ParentT> WithConflictResolutionPolicy(ConflictResolutionPolicy conflictResolutionPolicy);
+    }
+
+    /// <summary>
+    /// The stage of a sql container definition allowing to specify child resource.
+    /// </summary>
+    /// <typeparam name="ParentT">The stage of the parent definition to return to after attaching this definition.</typeparam>
+    public interface IWithChildResource<ParentT>
+    {
+        /// <summary>
+        /// Specifies a stored procedure.
+        /// </summary>
+        /// <param name="name">The name of the stored procedure.</param>
+        /// <param name="resource">The store procedure resource, no need to specify id.</param>
+        /// <param name="options">The options for the store procedure.</param>
+        /// <returns>The next stage of the definition.</returns>
+        IWithAttach<ParentT> WithStoredProcedure(string name, SqlStoredProcedureResource resource = default(SqlStoredProcedureResource), IDictionary<string, string> options = default(IDictionary<string, string>));
```

Comment:
Double check whether the resource object is required. It is quite meaningless with only a name.